### PR TITLE
Classifier: Refactor survey task Chooser to conditionally show filter components

### LINKED ISSUE #3689: Survey task Filter button shows if no filters
## Package
- lib-classifier

## Describe the bug
When there aren't filters the Filter button still shows.

## To Reproduce
1. Go to https://frontend.preview.zooniverse.org/projects/elwest/woodpecker-cavity-cam/classify/workflow/16998?env=demo
2. Click Filters button
3. Note there are no filters

## Expected behavior
If the survey task doesn't have any filters, don't show a Filter button.

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-classifier

## Linked Issue and/or Talk Post
Closes #3689 , towards other accessibility issues

## Describe your changes
- conditionally render survey task filter components (FilterStatus and ClearFilters) if there are filters
- update tests with react-testing-library

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
  - with filters - https://local.zooniverse.org:8080/?project=markb-panoptes%2Fsurvey-task-test-project&workflow=2434
  - without filters - https://local.zooniverse.org:8080/?project=markb-panoptes%2Fsurvey-task-test-project&workflow=3627
- What user actions should my reviewer step through to review this PR?
  - confirm filters components are shown or not shown per survey task with or without filters
- Which storybook stories should be reviewed? n/a
- Are there plans for follow up PR’s to further fix this bug or develop this feature? yes, see linked Milestone

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Bug Fix
- [ ] The PR creator has listed user actions to use when testing if bug is fixed
- [ ] The bug is fixed
- [ ] Unit tests are added or updated

### CODE REVIEW

**Comment by mcbouslog** [comment]
test timeout

File: packages/lib-classifier/src/plugins/tasks/survey/components/components/Chooser/Chooser.js [hunk 1]
```diff
@@ -37,11 +40,13 @@ function Chooser({
...
-        showingChoices={filteredChoiceIds.length}
-        totalChoices={task.choicesOrder?.length}
-      />
+      {showFilters
+        ? (<ClearFilters
+            handleFilter={handleFilter}
+            showingChoices={filteredChoiceIds.length}
+            totalChoices={task.choicesOrder?.length}
+           />)
+        : null}
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/plugins/tasks/survey/components/components/Chooser/Chooser.js:49]: Components can wrap other components, so the filters could be wrapped up into something like:
```jsx
<Filters>
  <Choices>
</Filters>
```
where `Filters` could contain the logic for showing the filters.

File: packages/lib-classifier/src/plugins/tasks/survey/components/components/Chooser/Chooser.spec.js [hunk 2]
```diff
@@ -1,37 +1,75 @@
...
+      render(
+        <Chooser
+          task={mockTask}
+        />
+      )
+      expect(screen.getByText('Aardvark')).to.be.ok() // first choice per mock-data survey task
+      expect(screen.getByText('Nothing here')).to.be.ok() // last choice per mock-data survey task
+    })
+
+    it('should allow you to filter the list of choices', function () {
```
Comment (eatyourgreens) [inline packages/lib-classifier/src/plugins/tasks/survey/components/components/Chooser/Chooser.spec.js:57]: Should there also be a user event test that selects a filter and filters the list of choices?

**Review by eatyourgreens — APPROVED** [review]
LGTM.

Could maybe use a test that selects a filter an applies it.

File: packages/lib-classifier/src/plugins/tasks/survey/components/components/Chooser/Chooser.spec.js (hunk 2 above)
Comment (eatyourgreens) [reply packages/lib-classifier/src/plugins/tasks/survey/components/components/Chooser/Chooser.spec.js:57]: I can’t see any tests that check the output of `<Choices />` after `FilterStatus` changes.

Comment (mcbouslog) [reply packages/lib-classifier/src/plugins/tasks/survey/components/components/Chooser/Chooser.spec.js:57]: Great suggestion and I agree that Chooser should have a user event test related to choices after a filter is selected, but I'm running into issues with the filter buttons that I'm working on fixing in #3735 . 

For now I added a test when a filter is provided as a prop to the Chooser component, that confirms the choices are as expected, and I'll add a comment to #3735 to further refactor the Chooser tests with the refactored filtered buttons.

**Pull request merged**